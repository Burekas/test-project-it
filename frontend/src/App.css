.App {
  margin: auto;
  margin-top: 50px;
  width: 40%;
  border: 1px dotted black;
  padding: 20px
}

.App-header {
  padding: 10px;
  text-align: center;
}

.App-title {
  font-size: 1.5em;
}
