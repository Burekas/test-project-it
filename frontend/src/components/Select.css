.Select, .Select-label {
    display: block;
    width: 100%;
    font-size: 1em;
}

.Select {
    margin-top: 5px;
    margin-bottom: 5px;
}
