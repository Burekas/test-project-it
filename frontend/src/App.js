import React from 'react';
import './App.css';
import TreeToSelects from './components/TreeToSelects';


const baseUrl = "http://127.0.0.1:8080"
const rootNodesUrl = `${baseUrl}/api/v1/get-root-nodes`;
// get-child-nodes/parentId
const childNodesUrl = `${baseUrl}/api/v1/get-child-nodes/`;
// get-node-data/id
const nodeDataUrl = `${baseUrl}/api/v1/get-node-data/`
// Specify labels for tree, index == depth of node
const labels = ['Education Standard', 'Grade Level', 'Learning Domain', 'Alignment Tag']

const App = () => (
  <div className="App">
    <header className="App-header">
      <h1 className="App-title">Test Project IT</h1>
    </header>
    <TreeToSelects rootNodesUrl={rootNodesUrl} childNodesUrl={childNodesUrl}
      nodeDataUrl={nodeDataUrl} labels={labels} />
  </div>
)

export default App;
