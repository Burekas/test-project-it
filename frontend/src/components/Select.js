import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Select.css';


/** Select component */
export default class Select extends Component {

  constructor(props) {
      super(props);
      this.handleChange = this.handleChange.bind(this);
  }

  /** call parent handler with id of selected option and select depth */
  handleChange(event) {
    this.props.onChange(event.target.value, this.props.depth);
  }

  render() {
    return (
      <div>
        <label className="Select-label">{this.props.label}
          <select className="Select" defaultValue="" onChange={this.handleChange}>
            <option value=""> -- Select an option -- </option>
            {this.props.options.map((option) =>
              <option key={option._id} value={option._id}>{option.name}</option>
            )}
          </select>
        </label>
      </div>
    )
  }

}

Select.propTypes = {
  /** seelct options */
  options: PropTypes.array.isRequired,
  /** change handler */
  onChange: PropTypes.func.isRequired,
  /** Depth of the select and nodes */
  depth: PropTypes.number.isRequired,
  /** Select label */
  label: PropTypes.string.isRequired
}
