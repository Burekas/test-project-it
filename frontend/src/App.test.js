import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from './App';
import Select from './components/Select';

Enzyme.configure({ adapter: new Adapter() });

describe('App', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const comp = renderer.create(<App />);
    const tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


describe('Select', () => {
    const props = {
        options: [
         {_id: '1', name: 'First'},
         {_id: '2', name: 'Second'},
        ],
        onChange: () => {},
        depth: 1,
        label: 'Label'
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Select { ...props } />, div);
    });

    it('shows tree options', () => {
        const element = shallow(<Select { ...props } />);
        /** third option is empty */
        expect(element.find('option').length).toBe(3);
    });

    test('has a valid snapshot', () => {
      const comp = renderer.create(<Select { ...props } />);
      const tree = comp.toJSON();
      expect(tree).toMatchSnapshot();
    });
});
