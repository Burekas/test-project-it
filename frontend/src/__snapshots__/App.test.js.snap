// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Test Project IT
    </h1>
  </header>
  <div>
    <div />
  </div>
</div>
`;

exports[`Select has a valid snapshot 1`] = `
<div>
  <label
    className="Select-label"
  >
    Label
    <select
      className="Select"
      defaultValue=""
      onChange={[Function]}
    >
      <option
        value=""
      >
         -- Select an option -- 
      </option>
      <option
        value="1"
      >
        First
      </option>
      <option
        value="2"
      >
        Second
      </option>
    </select>
  </label>
</div>
`;
